prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -91,48 +93,55 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
+                            reportId,
+                            jksCredential,
+                            credential,
+                            logger
+                        )
+                        logger.info("Token successfully added!")
 
-                    // post the report
-                    val reportClient = httpClient ?: createDefaultHttpClient(jksCredential, bearerTokens)
-                    val response = postReport(
-                        reportContent,
-                        fileName,
-                        restTransportInfo.reportUrl,
-                        httpHeaders,
-                        logger,
-                        reportClient
-                    )
-                    val responseBody = response.bodyAsText()
-                    // update the action history
-                    val msg = "Success: REST transport of $fileName to $restTransportInfo:\n$responseBody"
-                    logger.info("Message successfully sent!")
-                    actionHistory.trackActionResult(response.status, msg)
-                    actionHistory.trackSentReport(
-                        receiver,
-                        sentReportId,
-                        fileName,
-                        restTransportInfo.toString(),
-                        msg,
-                        header.reportFile.itemCount
-                    )
-                    actionHistory.trackItemLineages(Report.createItemLineagesFromDb(header, sentReportId))
+                        // post the report
+                        val reportClient = httpClient ?: createDefaultHttpClient(jksCredential, bearerTokens)
+                        val response = postReport(
+                            reportContent,
+                            fileName,
+                            restTransportInfo.reportUrl,
+                            httpHeaders,
+                            logger,
+                            reportClient
+                        )
+                        val responseBody = response.bodyAsText()
+                        // update the action history
+                        val msg = "Success: REST transport of $fileName to $restTransportInfo:\n$responseBody"
+                        logger.info("Message successfully sent!")
+                        actionHistory.trackActionResult(response.status, msg)
+                        actionHistory.trackSentReport(
+                            receiver,
+                            sentReportId,
+                            fileName,
+                            restTransportInfo.toString(),
+                            msg,
+                            header.reportFile.itemCount
+                        )
+                        actionHistory.trackItemLineages(Report.createItemLineagesFromDb(header, sentReportId))
+                    } catch (t: Throwable) {
+                        throw t
+                    }
                 }
```

Comment:
Why do we need to surround this block with another try block? Can we add some logging inside the catch?

Reply:
This is to allow the exception throw to cascade to the outside catch at line 149.  Therefore, we can use the very same catch to handle the exception.  Otherwise, it won't get to the catch block.